numpy/__init__.pyi

```diff
@@ -293,84 +399,20 @@ class ndarray(_ArrayOrScalarCommon, Iterable, Sized, Container):
-    @overload
-    def item(self, *args: int) -> Any: ...
-    @overload
-    def item(self, args: Tuple[int, ...]) -> Any: ...
-    def tolist(self) -> List[Any]: ...
-    @overload
-    def itemset(self, __value: Any) -> None: ...
-    @overload
-    def itemset(self, __item: _ShapeLike, __value: Any) -> None: ...
-    def tobytes(self, order: _OrderKACF = ...) -> bytes: ...
-    def tofile(
-        self, fid: Union[IO[bytes], str], sep: str = ..., format: str = ...
-    ) -> None: ...
-    def dump(self, file: str) -> None: ...
-    def dumps(self) -> bytes: ...
-    def astype(
-        self: _ArraySelf,
-        dtype: DtypeLike,
-        order: _OrderKACF = ...,
-        casting: _Casting = ...,
-        subok: bool = ...,
-        copy: bool = ...,
-    ) -> _ArraySelf: ...
-    def byteswap(self: _ArraySelf, inplace: bool = ...) -> _ArraySelf: ...
-    def copy(self: _ArraySelf, order: _OrderKACF = ...) -> _ArraySelf: ...
-    @overload
-    def view(self, type: Type[_NdArraySubClass]) -> _NdArraySubClass: ...
-    @overload
-    def view(self: _ArraySelf, dtype: DtypeLike = ...) -> _ArraySelf: ...
-    @overload
-    def view(
-        self, dtype: DtypeLike, type: Type[_NdArraySubClass]
-    ) -> _NdArraySubClass: ...
-    def getfield(
-        self: _ArraySelf, dtype: DtypeLike, offset: int = ...
-    ) -> _ArraySelf: ...
-    def setflags(
-        self, write: bool = ..., align: bool = ..., uic: bool = ...
-    ) -> None: ...
-    def fill(self, value: Any) -> None: ...
-    # Shape manipulation
-    @overload
-    def reshape(
-        self: _ArraySelf, shape: Sequence[int], *, order: _OrderACF = ...
-    ) -> _ArraySelf: ...
-    @overload
-    def reshape(
-        self: _ArraySelf, *shape: int, order: _OrderACF = ...
-    ) -> _ArraySelf: ...
-    @overload
-    def resize(self, new_shape: Sequence[int], *, refcheck: bool = ...) -> None: ...
-    @overload
-    def resize(self, *new_shape: int, refcheck: bool = ...) -> None: ...
-    @overload
-    def transpose(self: _ArraySelf, axes: Sequence[int]) -> _ArraySelf: ...
-    @overload
-    def transpose(self: _ArraySelf, *axes: int) -> _ArraySelf: ...
-    def swapaxes(self: _ArraySelf, axis1: int, axis2: int) -> _ArraySelf: ...
-    def flatten(self: _ArraySelf, order: _OrderKACF = ...) -> _ArraySelf: ...
-    def ravel(self: _ArraySelf, order: _OrderKACF = ...) -> _ArraySelf: ...
-    def squeeze(
-        self: _ArraySelf, axis: Union[int, Tuple[int, ...]] = ...
-    ) -> _ArraySelf: ...
     # Many of these special methods are irrelevant currently, since protocols
     # aren't supported yet. That said, I'm adding them for completeness.
     # https://docs.python.org/3/reference/datamodel.html
     def __len__(self) -> int: ...
-    def __getitem__(self, key) -> Any: ...
     def __setitem__(self, key, value): ...
     def __iter__(self) -> Any: ...
     def __contains__(self, key) -> bool: ...
     def __index__(self) -> int: ...
+    def __matmul__(self, other): ...
```

Comment:
`__matmul__` and the likes is only defined for `np.ndarray`; not `np.generic`

Reply:
If you're doing this type of cleanup... All the `__i<op>` methods are also present only on arrays.

Reply:
Might as well while we're at it; done as of https://github.com/numpy/numpy/pull/17188/commits/478c3f4ba320e580673da8a97b57ef45d96dac4a.